src/info.rs

```diff
@@ -187,6 +193,64 @@ impl ConnectionInfo {
     }
 }
 
+impl FromRequest for ConnectionInfo {
+    type Error = Infallible;
+    type Future = Ready<Result<ConnectionInfo, Infallible>>;
+    type Config = ();
+
+    fn from_request(req: &HttpRequest, _: &mut Payload) -> Self::Future {
+        ok(req.connection_info().clone())
+    }
+}
+
+/// Extractor for peer's socket address.
+///
+/// Also see [`HttpRequest::peer_addr`].
+///
+/// # Examples
+/// ```
+/// # use actix_web::Responder;
+/// use actix_web::dev::PeerAddr;
+///
+/// async fn handler(peer_addr: PeerAddr) -> impl Responder {
+///     let socket_addr = peer_addr.0;
+///     socket_addr.to_string()
+/// }
+/// # let _svc = actix_web::web::to(handler);
+/// ```
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord, Display)]
+#[display(fmt = "{}", _0)]
+pub struct PeerAddr(pub SocketAddr);
+
+impl PeerAddr {
+    /// Unwrap into inner `SocketAddr` value.
+    pub fn into_inner(self) -> SocketAddr {
+        self.0
+    }
+}
+
+#[derive(Debug, Display, Error)]
+#[display(fmt = "Missing peer address")]
+pub struct MissingPeerAddr;
```

Comment:
Is there any point in allowing users to construct this type? If not, I'd suggest making it `#[non_exhaustive]`.

Reply:
this is such a hot tip, i will remember this forever; i'd missed the nuance of non_exhaustive that it made structs non contstructible

Reply:
Another tip about this: You can add `#![warn(clippy::exhaustive_structs)]` to the crate attributes to catch all instances of types being constructible by users using struct literals and selectively add `#[allow(clippy::exhaustive_structs)]` where appropriate.